controllers/snapshot/snapshot_adapter.go

```diff
@@ -154,11 +155,20 @@ func (a *Adapter) EnsureGlobalCandidateImageUpdated() (reconciler.OperationResul
 		for _, component := range a.snapshot.Spec.Components {
 			if component.Name == a.component.Name {
 				a.component.Spec.ContainerImage = component.ContainerImage
-				err := a.client.Patch(a.context, a.component, patch)
-				if err != nil {
-					a.logger.Error(err, "Failed to update Global Candidate for the Component",
+				err1 := a.client.Patch(a.context, a.component, patch)
+				if err1 != nil {
+					a.logger.Error(err1, "Failed to update Global Candidate for the Component",
 						"Component.Name", a.component.Name)
-					return reconciler.RequeueWithError(err)
+					return reconciler.RequeueWithError(err1)
+				}
+				if reflect.ValueOf(component.Source).IsValid() {
```

Comment:
I feel we should also check if the value of `component.Source.GitSource.Revision` exist here